### PR TITLE
Migrate Microsoft.Bot.Builder.Dialogs.Declarative.Tests project to xUnit

### PR DESCRIPTION
**Pull request by ceciliaavila**
Fixes# 4095

## Description
This PR migrates all [Microsoft.Bot.Builder.Dialogs.Declarative.Tests](https://github.com/microsoft/botbuilder-dotnet/tree/main/tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests) tests from **MSTest** to **xUnit**.

## Detailed Changes
- Removed MSTest packages from the .csproj.
- Replaced `[TestMethod]` with `[Fact]`.
- Changed Asserts.
- Replaced `Assert.Fail` with `thown new XunitException`.
- Removed `TestContext`.
- Replaced `ClassInitialize` method with a class' constructor to execute tests' shared code.
- Used constants instead of variables where applicable.
- Used `var` instead of the specific type.
- Removed redundant parenthesis.

Also, we did the following changes in: 
- **_JsonLoadTests_**:
    - Added the `Dispose` method to clean up after each test execution.
    - Added the test name as parameter in `BuildTestFlow` and `BuildQnAMakerTestFlow` to pass it to `InitializeAdapter` method instead of `TestContext.TestName`
- **_SchemaMergeTests_**:
   - Replaced `[DataTestMethod]` with `Theory`
- **_Startup_**:
  - Replaced `[AssemblyInitialize]` with the `AssemblyFixture` feature of Xunit. 

## Testing
The following image shows the tests passing after the changes.
image

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/ResourceTests.cs [hunk 1]
```diff
@@ -9,22 +9,14 @@
...
 {
-    [TestClass]
     public class ResourceTests
     {
-        private static JsonSerializerSettings jsonSerializerSettings =
-            new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore, Formatting = Formatting.Indented };
-
-        public TestContext TestContext { get; set; }
-
-        [ClassInitialize]
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/ResourceTests.cs:26]: I think we should use IClassFixture replacing ClassInitialize so it is executed once before running all the tests in this class.

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/SchemaMergeTests.cs [hunk 2]
```diff
@@ -25,19 +26,9 @@ namespace Microsoft.Bot.Builder.Dialogs.Declarative.Tests
...
     {
-        public static ResourceExplorer ResourceExplorer { get; set; }
-
-        public static JSchema Schema { get; set; }
-
-        public static IEnumerable<object[]> Dialogs { get; set; }
-
-        public TestContext TestContext { get; set; }
-
-        [ClassInitialize]
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/SchemaMergeTests.cs:39]: I think we should use IClassFixture replacing ClassInitialize so it is executed once before running all the tests in this class.

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadTests.cs [hunk 3]
```diff
@@ -11,46 +11,41 @@
...
-    [TestClass]
-    public class JsonLoadTests
+    public class JsonLoadTests : IDisposable
     {
-        private static ResourceExplorer resourceExplorer;
+        private static ResourceExplorer _resourceExplorer;
 
-        public TestContext TestContext { get; set; }
-
-        [ClassInitialize]
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadTests.cs:25]: I think we should use IClassFixture replacing ClassInitialize so it is executed once before running all the tests in this class.

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/ResourceTests.cs (hunk 1 above)
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/ResourceTests.cs:26]: Feedback applied. Thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadTests.cs (hunk 3 above)
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadTests.cs:25]: Feedback applied. Thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/SchemaMergeTests.cs (hunk 2 above)
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/SchemaMergeTests.cs:39]: I couldn't implement _IClassFixture_ in this class as it needs a static constructor.

**Review by sw-joelmut — APPROVED** [review]

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadTests.cs [hunk 4]
```diff
@@ -327,96 +313,96 @@ await BuildTestFlow("HttpRequest.main.dialog")
...
 
-        [TestMethod]
+        [Fact]
         public async Task JsonDialogLoad_QnAMakerDialog_ActiveLearning_WithNoResponse()
         {
             var suggestionList = new List<string> { "Q1", "Q2", "Q3" };
             var suggestionActivity = QnACardBuilder.GetSuggestionsCard(suggestionList, "Did you mean:", "None of the above.");
             var qnAMakerCardEqualityComparer = new QnAMakerCardEqualityComparer();
-            var noAnswerActivity = "Answers not found in kb.";
+            const string noAnswerActivity = "Answers not found in kb.";
```
Comment (santgr11) [inline tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadTests.cs:337]: When the assigned object is explicit shouldn't go var? or this changes when its a const?

Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadTests.cs:337]: Var is for variables, if the value doesn't change it must be a const.

**Review by santgr11 — APPROVED** [review]

**Review by iscaltritti-sw — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**